Experimental Feature: Game Screenshots

LitRitt opened the pull request.

LitRitt committed:
Adds extension to handle interactions with Photos app
- isAuthorized to check/ask for authorization
- saveUIImage to save a UIImage to Photos

LitRitt committed:
Adds a Pause Menu button for taking game screenshots

LitRitt committed:
Adds @Feature and @Options for Game Screenshots

LitRitt committed:
Implements Game Screenshots feature in GameViewController

LitRitt committed:
Updates project settings

rileytestut commented on Delta/Emulation/GameViewController.swift:
```diff
@@ -1070,6 +1074,71 @@ extension GameViewController
+        }
+
+        if ExperimentalFeatures.shared.gameScreenshots.saveToPhotos,
+           PHPhotoLibrary.isAuthorized
```
See my note below, this is a potential bug.

rileytestut reviewed (changes requested):
Thanks for breaking these PRs up! Overall looks good to me, just have one note about the implementation of `isAuthorized` that could result in minor bug, but once that’s fixed I’d be happy to merge 🙂

Also I appreciate you putting the PHPhotoLibrary stuff in an extension! The more isolated changes are from the rest of the app, the better 👍

LitRitt committed:
Passes call to save to Photos as a closure into authorization prompt
- Ensures that the screenshot is saved when the user is first prompted for access to Photos
- More elegant extension code

rileytestut reviewed (approved):
Nice approach, passing in a closure that's only run if authorized is a good idea. 

Thanks for these changes, everything looks good to me! Will merge this in along with #241 once everything is merged into `develop` branch 👍

rileytestut committed:
Merge branch 'experimental_features' into experimental_feature/game_screenshots

takashiou reviewed (commented):
好👍

jesn459 reviewed (commented):
Gracias
